java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -157,10 +157,10 @@ public static String getServerDataFile() {
         File standaloneDir = new File(METADATA_DIR_STANDALONE_SALT);
         File managerDir = new File(METADATA_DIR_MANAGER);
         File customDir = new File(METADATA_DIR_CUSTOM);
-        List<File> files = new LinkedList<>(
-                Arrays.asList(standaloneDir.listFiles()));
-        files.addAll(Arrays.asList(managerDir.listFiles()));
-        files.addAll(Arrays.asList(customDir.listFiles()));
+        List<File> files = new LinkedList<>();
+        Optional.ofNullable(standaloneDir.listFiles()).ifPresent(l-> files.addAll(Arrays.asList(l)));
```

Comment:
just nitpicking: to be more readable I would change the local function variable "l" to standaloneFiles.

```java
Optional.ofNullable(standaloneDir.listFiles()).ifPresent(standaloneFiles -> files.addAll(Arrays.asList(standaloneFiles)));
Optional.ofNullable(managerDir.listFiles()).ifPresent(managerFiles-> files.addAll(Arrays.asList(managerFiles)));
```